app/models/nic/bond.rb

```diff
@@ -44,6 +45,17 @@ def ensure_virtual
     def enc_attributes
       @enc_attributes ||= (super + %w(mode attached_devices bond_options))
     end
+
+    def ensure_mac
+      mac_addresses = attached_devices_identifiers.collect { |a|
```

Comment:
perhaps this would be better
`mac_addresses = self.host.interfaces.where(:identifier => attached_devices_identifiers).pluck(:mac)`